internal/common/typescript_mock_lib.bzl

```diff
@@ -19,19 +19,42 @@ rules_typescript without introducing a circular dependency between
 rules_nodejs and rules_typescript repositories.
 """
 
+load("@build_bazel_rules_nodejs//:declaration_provider.bzl", "DeclarationInfo")
+load("@build_bazel_rules_nodejs//:providers.bzl", "JSEcmaScriptModuleInfo")
+
 def _mock_typescript_lib(ctx):
```

Comment:
opt: we mocked typescript because the built-in couldn't depend on real ts_library but maybe now it should?

Reply:
I think so too. I'll rip out mock_typescript_lib in a pre-factor PR

Reply:
Done

Reply:
in #1212